src/tests/multihost/ipa/test_misc.py

```diff
@@ -430,3 +430,89 @@ def check_hostname_hash(hash_value=None):
+        add_anony_pkinit = {'krb5_fast_use_anonymous_pkinit': 'True'}
+        sssd_client.sssd_conf(domain_name, add_anony_pkinit)
+        sssd_client.clear_sssd_cache()
+        cmd_kinit = multihost.client[0].run_command('kinit -n')
+        assert cmd_kinit.returncode == 0
+        ssh = SSHClient(multihost.client[0].ip,
+                        username='foobar0', password='Secret123')
+        ssh.close()
+        cmd_klist = 'klist /var/lib/sss/db/fast_ccache_%s' % \
+                    sssd_client.get_domain_section_name().upper()
+        output = multihost.client[0].run_command(cmd_klist).stdout_text
+        principal = 'WELLKNOWN/ANONYMOUS@WELLKNOWN:ANONYMOUS'
+        assert principal in output
+
+    def test_anonymous_pkinit_for_fast_false(self, multihost, backupsssdconf):
+        """
+        :title: Negative test for allow SSSD to use anonymous pkinit for FAST
+        :id: de823122-af88-41f6-b762-63083fccaa87
+        :customerscenario: True
+        :description:
+         For SSSD to use FAST a Kerberos keytab and service principal must
+         exist. SSSD to be enhanced to allow for the use of anonymous pkinit
+         to create the FAST session. With
+         'krb5_fast_use_anonymous_pkinit = False' the ccache will have a
+         ticket for the host principal.
+        :steps:
+          1. Setup a IPA server/client with default setting.
+          2. Call anonymous processing using #kinit -n.
+          3. Set 'krb5_fast_use_anonymous_pkinit = False' in sssd.conf.
+          4. Login to the IPA user.
+          5. Check a ccache file for the host principal.
+        :expectedresults:
+          1. Successfully setup the IPA server/client.
+          2. Successfully called anonymous processing.
+          3. Successfully set the option in sssd.conf.
+          4. Successfully logged in to IPA user.
+          5. Successfully get a ccache file with the host principal.
+        :bugzilla:
+        https://bugzilla.redhat.com/show_bug.cgi?id=1859751
+        """
+        sssd_client = sssdTools(multihost.client[0])
+        domain_section = sssd_client.get_domain_section_name()
+        domain_name = '%s/%s' % ('domain', domain_section)
+        add_anony_pkinit = {'krb5_fast_use_anonymous_pkinit': 'False'}
+        sssd_client.sssd_conf(domain_name, add_anony_pkinit)
+        sssd_client.clear_sssd_cache()
+        cmd_kinit = multihost.client[0].run_command('kinit -n')
+        assert cmd_kinit.returncode == 0
+        ssh = SSHClient(multihost.client[0].ip,
+                        username='foobar1', password='Secret123')
+        ssh.close()
+        cmd_klist = 'klist /var/lib/sss/db/fast_ccache_%s' % \
+                    domain_section.upper()
+        output = multihost.client[0].run_command(cmd_klist).stdout_text
+        principal = re.compile(
+            r'principal:.host.%s@%s' % multihost.client[0].sys_hostname,
+            domain_section.upper())
```

Comment:
Hi,

there are missing braces around the parameters:
```
diff --git a/src/tests/multihost/ipa/test_misc.py b/src/tests/multihost/ipa/test_misc.py
index 101ef6a39..efdb7e02f 100644
--- a/src/tests/multihost/ipa/test_misc.py
+++ b/src/tests/multihost/ipa/test_misc.py
@@ -513,6 +513,6 @@ class Testipabz(object):
                     domain_section.upper()
         output = multihost.client[0].run_command(cmd_klist).stdout_text
         principal = re.compile(
-            r'principal:.host.%s@%s' % multihost.client[0].sys_hostname,
-            domain_section.upper())
+            r'principal:.host.%s@%s' % (multihost.client[0].sys_hostname,
+            domain_section.upper()))
         assert principal.search(output)
```

With this expansion I think you can also use `assert principal in output` with `principal = 'principal: host/%s@%s' % (multihost.client[0].sys_hostname, domain_section.upper())'`. But using a regular expression is fine as well.

bye,
Sumit